core/templates/dev/head/pages/email_dashboard/EmailDashboard.js

```diff
@@ -0,0 +1,121 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Controller for oppia email dashboard page.
+ */
+
+oppia.controller('EmailDashboard', [
+  '$scope', '$http', function($scope, $http) {
+    var QUERY_DATA_URL = '/emaildashboarddatahandler';
+
+    // Store current cursor.
+    $scope.currentCursor = null;
```

Comment:
You suggest that we keep only latest cursor and all the rest of queries should be cached but what about Reload button. If we use such data structure user won't be able to use Reload button. Reloading any set of queries from backend will require cursor of that set.

So if we need Reload button then we have to store cursor + queries. In addition to that we will also need next_cursor obtained during last fetch. So in that case we need some thing like repeated dictionaries.
```
$scope.queries = [ {
    queries: (queries of particular page)
    cursor: (cursor for reloading particular page)
 } for each page fetched from backend]
```

Your approach works if reload button is not present. I am fine with either way. We completely remove Reload button or we use something like above to keep it.